Core/src/org/sleuthkit/autopsy/datamodel/Accounts.java

```diff
@@ -764,14 +786,17 @@ public long getHits() {
         protected boolean createKeys(List<FileWithCCN> list) {
             String query
                     = "SELECT blackboard_artifacts.obj_id," //NON-NLS
-                    + "      blackboard_attributes.value_text AS solr_document_id, " //NON-NLS
+                    + "      solr_attribute.value_text AS solr_document_id, " //NON-NLS
                     + "      GROUP_CONCAT(blackboard_artifacts.artifact_id) AS artifact_IDs, " //NON-NLS
                     + "      COUNT( blackboard_artifacts.artifact_id) AS hits,  " //NON-NLS
                     + "      GROUP_CONCAT(blackboard_artifacts.review_status_id) AS review_status_ids "
                     + " FROM blackboard_artifacts " //NON-NLS
-                    + " LEFT JOIN blackboard_attributes ON blackboard_artifacts.artifact_id = blackboard_attributes.artifact_id " //NON-NLS
-                    + "                                AND blackboard_attributes.attribute_type_id = " + BlackboardAttribute.ATTRIBUTE_TYPE.TSK_SOLR_DOCUMENT_ID.getTypeID() //NON-NLS
-                    + " WHERE blackboard_artifacts.artifact_type_id = " + BlackboardArtifact.ARTIFACT_TYPE.TSK_CREDIT_CARD_ACCOUNT.getTypeID() //NON-NLS
+                    + " LEFT JOIN blackboard_attributes as solr_attribute ON blackboard_artifacts.artifact_id = solr_attribute.artifact_id " //NON-NLS
+                    + "                                AND solr_attribute.attribute_type_id = " + BlackboardAttribute.ATTRIBUTE_TYPE.TSK_SOLR_DOCUMENT_ID.getTypeID() //NON-NLS
+                    + " LEFT JOIN blackboard_attributes as account_type ON blackboard_artifacts.artifact_id = account_type.artifact_id " //NON-NLS
```

Comment:
Since there is some chance that we will be using Elastic instead of Solr in the future, I would prefer something like TSK_EXTRACTED_TXT_DOC_ID or TSK_KEYWORD_SEARCH_DOC_ID to TSK_SOLR_DOC_ID.

I am also wondering about TSK_PAYMENT_CARD_TYPE. There are several new TSK_CREDIT_CARD_ attributes, along with some more generic new attributes such as TSK_BRAND_(NAME?), TSK_BANK_NAME, TSK_COUNTRY, TSK_CITY. Is payment card type generic or part of the CREDIT_CARD attributes "family?